Sources/_MatchingEngine/Regex/AST/MatchingOptions.swift

```diff
@@ -53,6 +58,15 @@ extension AST {
         return false
       }
     }
+    
+    public var isSemanticMatchingLevel: Bool {
+      switch kind {
+      case .graphemeClusterSemantics, .unicodeScalarSemantics, .byteSemantics:
```

Comment:
The only existing engine I've found that has this semantic switching is Rust's `regex::bytes`, which lets you switch from the default byte-matching mode into a Unicode mode with `(?u)`.

Reply:
In Oniguruma, `.textSegmentGraphemeMode` and `.textSegmentWordMode` are spelled `y{g}` and `y{w}`, and change the behavior of `\y` and `\Y` between marking grapheme boundaries and marking word boundaries (as defined by https://unicode.org/reports/tr29/). That in turn changes the behavior of `\X`, since it consumes up to the next `\y` boundary.

Reply:
What are your thoughts there? This seems relevant to whether we should be using an enum in the AST as the same type/representation for our model.